homeassistant/components/group.py

```diff
@@ -161,48 +169,60 @@ def reload_service_handler(service_call):
     return True
 
 
+@callback
 def _process_config(hass, config, component):
     """Process group configuration."""
+    groups = []
     for object_id, conf in config.get(DOMAIN, {}).items():
         name = conf.get(CONF_NAME, object_id)
         entity_ids = conf.get(CONF_ENTITIES) or []
         icon = conf.get(CONF_ICON)
         view = conf.get(CONF_VIEW)
 
         group = Group(hass, name, entity_ids, icon=icon, view=view,
-                      object_id=object_id)
-        component.add_entities((group,))
+                      object_id=object_id, async=True)
+        groups.append(group)
 
-    return True
+    hass.async_add_job(component.async_add_entities, groups)
 
 
 class Group(Entity):
     """Track a group of entity ids."""
 
     # pylint: disable=too-many-instance-attributes, too-many-arguments
     def __init__(self, hass, name, entity_ids=None, user_defined=True,
-                 icon=None, view=False, object_id=None):
+                 icon=None, view=False, object_id=None, async=False):
```

Comment:
Don't name any variable `async`, that's a reserved keyword in Python 3.5+.

Also, it feels dirty to me that a constructor should be aware if it is async or not. In all other places in the code that we have a sync/async version, we have the sync version call the async version via `run_callback_threadsafe`.